src/Money.php

```diff
@@ -539,4 +477,29 @@ public function jsonSerialize()
             'currency' => $this->currency,
         );
     }
+
+    /**
+     * @param CalculatorInterface $calculator
+     */
+    public static function setCalculator(CalculatorInterface $calculator)
+    {
+        self::$calculator = $calculator;
+    }
+
+    /**
+     *
+     */
+    private static function assignCalculator()
+    {
+        /** @var CalculatorInterface[] $calculators */
+        $calculators = [new BcMathCalculator(), new GmpCalculator(), new PhpCalculator()];
```

Comment:
If the supported method would be static, we could even add possibility for custom calculators. Somthing like `Money::registerCalculator` with our calculators in the list out of the box.